File: mobile/android/LoHiTrade/app/src/main/java/com/lohitrade/data/models/AuthModels.kt
package com.lohitrade.data.models

import com.google.gson.annotations.SerializedName

// -- Request models --

data class LoginRequest(
    val email: String,
    val password: String
)

data class RegisterRequest(
    val email: String,
    val password: String,
    val phone: String,
    val name: String
)

data class GoogleLoginRequest(
    @SerializedName("id_token") val idToken: String
)

data class AppleLoginRequest(
    @SerializedName("auth_code") val authCode: String,
    @SerializedName("user_name") val userName: String? = null
)

data class RefreshTokenRequest(
    @SerializedName("refresh_token") val refreshToken: String
)

data class FCMTokenRequest(
    @SerializedName("fcm_token") val fcmToken: String
)

// -- Response models --

data class TokenResponse(
    @SerializedName("access_token") val accessToken: String,
    @SerializedName("refresh_token") val refreshToken: String,
    @SerializedName("token_type") val tokenType: String
)

data class RegisterResponse(
    @SerializedName("user_id") val userId: String,
    val email: String,
    val message: String
)

data class ErrorResponse(
    val detail: String
)

// -- JWT payload (decoded locally for expiry checks) --

data class JWTPayload(
    val sub: String,
    val email: String,
    val role: String,
    val type: String,
    val iat: Long,
    val exp: Long
) {
    /** Seconds remaining until token expires. */
    val secondsUntilExpiry: Long
        get() = exp - (System.currentTimeMillis() / 1000)

    /** Whether the token has expired. */
    val isExpired: Boolean
        get() = secondsUntilExpiry <= 0

    /** Whether the token will expire within the given threshold (default 60s). */
    fun expiresWithin(seconds: Long = 60): Boolean = secondsUntilExpiry < seconds
}

File: mobile/ios/LoHiTrade/LoHiTrade/Models/AuthModels.swift
import Foundation

// MARK: - Request models

struct LoginRequest: Encodable {
    let email: String
    let password: String
}

struct RegisterRequest: Encodable {
    let email: String
    let password: String
    let phone: String
    let name: String
}

struct GoogleLoginRequest: Encodable {
    let idToken: String

    enum CodingKeys: String, CodingKey {
        case idToken = "id_token"
    }
}

struct AppleLoginRequest: Encodable {
    let authCode: String
    let userName: String?

    enum CodingKeys: String, CodingKey {
        case authCode = "auth_code"
        case userName = "user_name"
    }
}

struct RefreshTokenRequest: Encodable {
    let refreshToken: String

    enum CodingKeys: String, CodingKey {
        case refreshToken = "refresh_token"
    }
}

// MARK: - Response models

struct TokenResponse: Decodable, Equatable {
    let accessToken: String
    let refreshToken: String
    let tokenType: String

    enum CodingKeys: String, CodingKey {
        case accessToken = "access_token"
        case refreshToken = "refresh_token"
        case tokenType = "token_type"
    }
}

struct RegisterResponse: Decodable {
    let userId: String
    let email: String
    let message: String

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case email, message
    }
}

struct ErrorResponse: Decodable {
    let detail: String
}

// MARK: - JWT payload (decoded locally for expiry checks)

struct JWTPayload {
    let sub: String
    let email: String
    let role: String
    let type: String
    let iat: Int
    let exp: Int

    /// Seconds remaining until token expires.
    var secondsUntilExpiry: TimeInterval {
        TimeInterval(exp) - Date().timeIntervalSince1970
    }

    /// Whether the token has expired.
    var isExpired: Bool { secondsUntilExpiry <= 0 }

    /// Whether the token will expire within the given threshold (default 60s).
    func expiresWithin(_ seconds: TimeInterval = 60) -> Bool {
        secondsUntilExpiry < seconds
    }
}
